### PR TITLE
[5.1] doc-serialization: exclude non-public decls when serializing module group info.

### PR DESCRIPTION
**Pull request by nkcsgexi**
This reduces the size of x86_64.swiftdoc about 10%.

rdar://48689311

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Linux platform

File: test/Serialization/group_info_diags.swift [hunk 1]
```diff
@@ -4,7 +4,16 @@
 // RUN: not %target-swift-frontend -emit-module %s -module-name HasArray -o %t -module-cache-path %t/mcp -group-info-path %S/Inputs/corrupted_group_info.json -emit-module-doc &> %t/result.txt
 // RUN: %FileCheck %s -check-prefix=CORRUPTED < %t/result.txt
 
-public protocol P {}
+// RUN: %target-swift-frontend -emit-module %s -module-name HasArray -o %t -module-cache-path %t/mcp -group-info-path %S/Inputs/group_info.json -emit-module-doc &> %t/result.txt
+// RUN: strings %t/HasArray.swiftdoc > %t/doc_strings.txt
```
Comment (akyrtzi) [inline test/Serialization/group_info_diags.swift]: @shahmishal @compnerd should use of `strings` binary in the test be guarded with `// REQUIRES: shell` ?

Comment (compnerd) [reply test/Serialization/group_info_diags.swift]: No, you should use `llvm-strings` instead.

Comment (akyrtzi) [reply test/Serialization/group_info_diags.swift]: Ah nice, I wasn't aware of this, @nkcsgexi could you make such a change ?

**Review by akyrtzi — CHANGES_REQUESTED** [review]

File: test/Serialization/group_info_diags.swift (hunk 1 above)
Comment (nkcsgexi) [reply test/Serialization/group_info_diags.swift]: sure, I have cherry-picked @compnerd 's https://github.com/apple/swift/pull/24092 here

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Pull request merged**